test/file01.ts

```diff
@@ -0,0 +1,45 @@
+import { CdnDto } from '@xsite-ui/common.cdn';
+import { buildImageSrc } from '@xsite-ui/common.image';
+import { CrossOffersChart } from '@xsite-ui/common.rapido';
+import { OfferProduct } from '@xsite-ui/fintech.shared-types';
+import { getProductIdentifier, isApplicationEntityDataType } from '@xsite-ui/fintech.utils';
+
+import { BucketProps } from '../types';
+
+export function getOffersCrossChartProps(
+  crossOffersChart: BucketProps['crossOffersChart'],
+  cdnData: CdnDto,
+) {
+  const {
+    title = '',
+    secondTitle = '',
+    subTitle = '',
+    numOfOffers = 3,
+    productListV3 = {},
+  } = crossOffersChart || {};
+
+  if (!isApplicationEntityDataType<CrossOffersChart>(crossOffersChart)) {
+    return null;
+  }
+
+  const { products } = productListV3 as unknown as { products: OfferProduct[] };
+
+  const offerProducts = products
+    ?.map((product) => {
+      return {
+        ...product,
+        productIdentifier: getProductIdentifier(product),
+        logoPath: buildImageSrc({ src: product?.logoPath, cdn: cdnData }),
+      };
```

Comment:
:robot: OpenAI

Consider extracting the logic in this map function into a separate named function for better readability and reusability.

35:
Consider using `Array.prototype.slice` instead of the `?.` operator to avoid returning `undefined` if `products` is `null` or `undefined`.

```suggestion
import { CdnDto } from '@xsite-ui/common.cdn';
import { buildImageSrc } from '@xsite-ui/common.image';
import { CrossOffersChart } from '@xsite-ui/common.rapido';
import { OfferProduct } from '@xsite-ui/fintech.shared-types';
import { getProductIdentifier, isApplicationEntityDataType } from '@xsite-ui/fintech.utils';

import { BucketProps } from '../types';

function mapProductToOfferProduct(product: OfferProduct, cdnData: CdnDto): OfferProduct {
  return {
    ...product,
    productIdentifier: getProductIdentifier(product),
    logoPath: buildImageSrc({ src: product.logoPath, cdn: cdnData }),
  };
}

export function getOffersCrossChartProps(
  crossOffersChart: BucketProps['crossOffersChart'],
  cdnData: CdnDto,
) {
  if (!isApplicationEntityDataType<CrossOffersChart>(crossOffersChart)) {
    return null;
  }

  const { 
    title = '', 
    secondTitle = '', 
    subTitle = '', 
    numOfOffers = 3, 
    productListV3 = {}, 
  } = crossOffersChart;

  const { products } = productListV3;
  
  const offerProducts = products
    ? products.map((product) => mapProductToOfferProduct(product, cdnData)).slice(0, numOfOffers)
    : [];

  return {
    entityContent: {
      title,
      secondTitle,
      subTitle,
    },
    products: offerProducts,
  };
}
```

<!-- This is an auto-generated comment by OpenAI -->